### ISSUE TITLE
How to configure Gatsby with node API?

### ISSUE THREAD

**Issue by moroshko** [gatsbyjs/gatsby#15650]
I'm trying to create a monorepo with Gatsby front end and node API.
Here is a simplified version of the folder structure:
```
my-monorepo
  api
    contact-us.js
  src
    pages
      contact-us.js
  gatsby-config.js
  package.json
  now.json
```

Once the user submits the Contact Us form on the `/contact-us` page it should hit the `/api/contact-us.js` endpoint.

I'm using ZEIT's [now](https://zeit.co/now) to deploy the application.

However, when I run `now dev` and access `/api/contact-us` I get:
> Gatsby.js development 404 page
> There's not a page yet at /api/contact-us
> Create a React.js component in your site directory at src/pages/api/contact-us.js and this page will automatically refresh to show the new page component you created.

How could I tell Gatsby to stop trying to load pages at `/api/**`?

I tried to add [`gatsby-plugin-exclude`](https://www.gatsbyjs.org/packages/gatsby-plugin-exclude/):
```
{
  resolve: "gatsby-plugin-exclude",
  options: {
    paths: ["/api/**"],
  },
}
```
but it doesn't seem to do anything.

What am I missing?

---

**`package.json`**
```
{
  ...
  "scripts": {
    "build": "gatsby build",
    "now-build": "npm run build",
    "now-dev": "gatsby develop -p $PORT"
  }
}
```

**`now.json`**
```
{
  "version": 2,
  "name": "my-monorepo",
  "builds": [
    {
      "src": "package.json",
      "use": "@now/static-build",
      "config": { "distDir": "public" }
    },
    {
      "src": "api/*.js",
      "use": "@now/node"
    }
  ],
  "routes": [
    {
      "src": "/api/(.*)",
      "dest": "/api/$1.js"
    }
  ]
}

```

**Update**

When the app is deployed using `now`, everything works as expected. Visiting `/api/contact-us` returns lambda's response. So, looks like it's a `now dev` specific issue.

**Update 2**

Repro [here](https://github.com/moroshko/gatsby-now-dev-issue).

**Comment by pieh** [comment]
It seems like providing `now-dev` scripts completely takes over `now dev` (and route is not being handled) - this is probably question to `now` folks (I couldn't find much information about this).

But to workaround this - you can add this code snippet to `gatsby-node.js` file:
```
exports.onCreateDevServer = ({ app }) => {
  app.use(`/api/:function`, (req, res) => {
    try {
      require(`./api/${req.params.function}`)(req, res)
    } catch {
      res.status(404).end()
    }
    res.end()
  })
}
```
This will tell gatsby's dev server to handle `/api/<apiName>` requests by using request handlers in your `api` directory

**Comment by moroshko** [comment]
Thanks @pieh for taking a look!
It was indeed a bug in now CLI, and it's fixed now in version 15.8.0.
